docs/rfcs/0007-export-direction.md

```diff
@@ -0,0 +1,221 @@
+# Direction of Flow component exports
+
+* Date: 2023-03-22
+* Author: Paulin Tpdev (@ptodev)
+* PR:
+* Status: Draft
+
+## Summary
+It may be confusing that component exports can be either the component's input or its output. At the moment Flow doesn't enforce either model. People are free to fix and match them using their best judgement. This RFC evaluates the pros and cons of each approach, and also states if it'd be worth it to change our current approach.
+
+## Outcome
+The users would like to see the following:
+
+- When reading a component's documentation, know what is an input and what is an output at a glance.
+- Is should be easy to tell which component's output can be piped to which component's input (composition).
+- Too much reliance on examples is a sign that the system is too hard to use. Browsing the documentation for a component's properties should be enough most of the time. 
+- Learning new components should be easy and feel familiar. It should not feel like a new experience.
+
+## Examples
+
+### Example 1: Prometheus scrape and remote write
+#### Example 1.1: How Flow works at the moment
+```
+prometheus.remote_write "staging" {
+  endpoint {
+    url = "http://mimir:9009/api/v1/push"
+  }
+}
+
+prometheus.scrape "demo1" {
+  targets = [{"__address__" = "127.0.0.1:12345"}]
+  forward_to = [prometheus.remote_write.staging.receiver]
+}
+
+prometheus.scrape "demo2" {
+  targets = [{"__address__" = "127.0.0.2:12345"}]
+  forward_to = [prometheus.remote_write.staging.receiver]
+}
+```
+
+#### Example 1.2: How Flow could work if we do not use exports as inputs
```

Comment:
Could we list potential solutions separately? I think this makes the document difficult to follow.